src/naive.h

```diff
@@ -0,0 +1,21 @@
+#ifndef NAIVE_H
+#define NAIVE_H
+
+// Checks sequence[seq_index:] for pattern. Returns index when exact pattern is found. Otherwise -1.
+int naivly_get_position_of_next_match(char *sequence, char *pattern, int seq_index, int pattern_len) {
+    // We define, that the empty string can be found everywhere
+    if (pattern_len == 0) return seq_index+1;
+    // Try to find instances of pattern in sequence
+    for (int i = seq_index; i < strlen(sequence); i++) {
```

Comment:
I don't know what the running time is here, because the C standard doesn't require `strlen(-)` to run in linear time. Compilers often optimise so if they know that you haven't modified `sequence` then `strlen(-)` is known after the first time you called it. This _typically_ requires that `sequences` is `const`, but compilers are pretty good at working out what you doing, and you are not writing to data through a pointer here, and I suspect that I will work it out.

That, of course, is pure luck. You shouldn't call `strlen()` in a loop, since it _can_ take time proportional to the length of the string you give it.